src/env/tock/storage.rs

```diff
@@ -362,23 +394,23 @@ impl TockUpgradeStorage {
         let write_range = ModRange::new(address, data.len());
         if self.contains_metadata(&write_range)? {
             let new_metadata = &data[self.metadata.start() - address..][..self.metadata.length()];
-            check_metadata::<TockEnv>(self, UPGRADE_PUBLIC_KEY, new_metadata)?;
+            check_metadata::<TockEnv<S>, S, C>(self, UPGRADE_PUBLIC_KEY, new_metadata)?;
```

Comment:
It looks like we are taking the default config for TockEnv here. Do we want to pass the config on from above?
```suggestion
            check_metadata::<TockEnv<S, C>, S, C>(self, UPGRADE_PUBLIC_KEY, new_metadata)?;
```